src/SocketSession.cpp

```diff
@@ -3,3 +3,125 @@
 // found in the LICENSE file.
 
 #include "SocketSession.h"
+
+typedef boost::asio::buffers_iterator<boost::asio::streambuf::const_buffers_type> iterator;
+std::pair<iterator, bool> match_end_socket(iterator begin, iterator end) {
+	int index = 0;
+	uint8_t fin= 0x00;
+	iterator i = begin;
+	//print hexa decimal of data
+	while (i != end) {
+		//std::cout << std::hex << (int)(*i)<<" ";
+		if (index == 0)
+			fin = (uint8_t)((*i) >> 7) & 0x01;
+		index++;
+		i++;
+	}
+	//fin = (uint8_t)((*i) >> 7) & 0x01;
+	std::cout << std::endl;
+	if (fin == 0x01){
+		return std::make_pair(i, true);
+	}
+	else{
+		return std::make_pair(i, false);
+	}	
+}
+
+void SocketSession::receiveMessage() {
+	std::cout << "receive" << std::endl;
+
+	buffer_.consume(buffer_.size());
```

Comment:
I would recommend cleaning the buffer out at time of use instead. Here you check to see if the buffer is empty.